src/twisted/words/protocols/irc.py

```diff
@@ -2643,16 +2656,22 @@ def connectionMade(self):
         if self.performLogin:
             self.register(self.nickname)
 
-    def dataReceived(self, data):
+    def dataReceived(self, data: bytes | str):
```

Comment:
:grimacing: This is always `bytes`!

```suggestion
    def dataReceived(self, data: bytes) -> None:
```

Reply:
Yes, you're totally right. Fixed.